src/main/java/org/folio/sidecar/service/PathProcessor.java

```diff
@@ -0,0 +1,63 @@
+package org.folio.sidecar.service;
+
+import static org.folio.sidecar.model.ModulePrefixStrategy.PROXY;
+import static org.folio.sidecar.model.ModulePrefixStrategy.STRIP;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.folio.sidecar.configuration.properties.ModuleProperties;
+import org.folio.sidecar.configuration.properties.SidecarProperties;
+import org.folio.sidecar.model.ModulePrefixStrategy;
+
+@ApplicationScoped
+public class PathProcessor {
+
+  private final ModuleProperties moduleProperties;
+  private final ModulePrefixStrategy modulePrefixStrategy;
+
+  /**
+   * Injects dependencies from quarkus context.
+   *
+   * @param moduleProperties - underlying module configuration
+   * @param scProperties - sidecar properties
+   */
+  @Inject
+  public PathProcessor(ModuleProperties moduleProperties, SidecarProperties scProperties) {
+    this.moduleProperties = moduleProperties;
+    this.modulePrefixStrategy = scProperties.isModulePrefixEnabled() ? PROXY : scProperties.getModulePrefixStrategy();
+  }
+
+  /**
+   * Builds path with {@code "/$moduleName"} prefix if it's defined by configuration.
+   *
+   * @param path - URL path
+   * @return updated path as {@link String} value with module name prefix (according to configuration)
+   */
+  public String getModulePath(String path) {
+    if (modulePrefixStrategy == PROXY) {
+      var modulePathPrefix = '/' + moduleProperties.getName();
```

Comment:
Looks like modulePathPrefix can be calculated once in a constructor, and then reused for this method and cleanIngressRequestPath